src/Control/Monad/Trans/Free.hs

```diff
@@ -212,80 +162,42 @@ free :: FreeF f a (Free f a) -> Free f a
 free = FreeT . Identity
 {-# INLINE free #-}
 
-#ifdef LIFTED_FUNCTOR_CLASSES
 instance (Eq1 f, Eq1 m, Eq a) => Eq (FreeT f m a) where
-#else
-instance (Functor f, Eq1 f, Functor m, Eq1 m, Eq a)=> Eq (FreeT f m a) where
-#endif
     (==) = eq1
 
-#ifdef LIFTED_FUNCTOR_CLASSES
 instance (Eq1 f, Eq1 m) => Eq1 (FreeT f m) where
   liftEq eq = go
     where
       go (FreeT x) (FreeT y) = liftEq (liftEq2 eq go) x y
-#else
-instance (Functor f, Eq1 f, Functor m, Eq1 m) => Eq1 (FreeT f m) where
-  eq1 = on eq1 (fmap (Lift1 . fmap Lift1) . runFreeT)
-#endif
 
-#ifdef LIFTED_FUNCTOR_CLASSES
 instance (Ord1 f, Ord1 m, Ord a) => Ord (FreeT f m a) where
-#else
-instance (Functor f, Ord1 f, Functor m, Ord1 m, Ord a) => Ord (FreeT f m a) where
-#endif
     compare = compare1
 
-#ifdef LIFTED_FUNCTOR_CLASSES
 instance (Ord1 f, Ord1 m) => Ord1 (FreeT f m) where
   liftCompare cmp = go
     where
       go (FreeT x) (FreeT y) = liftCompare (liftCompare2 cmp go) x y
-#else
-instance (Functor f, Ord1 f, Functor m, Ord1 m) => Ord1 (FreeT f m) where
-  compare1 = on compare1 (fmap (Lift1 . fmap Lift1) . runFreeT)
-#endif
 
-#ifdef LIFTED_FUNCTOR_CLASSES
 instance (Show1 f, Show1 m) => Show1 (FreeT f m) where
   liftShowsPrec sp sl = go
     where
       goList = liftShowList sp sl
       go d (FreeT x) = showsUnaryWith
         (liftShowsPrec (liftShowsPrec2 sp sl go goList) (liftShowList2 sp sl go goList))
         "FreeT" d x
-#else
-instance (Functor f, Show1 f, Functor m, Show1 m) => Show1 (FreeT f m) where
-  showsPrec1 d (FreeT m) = showParen (d > 10) $
-    showString "FreeT " . showsPrec1 11 (Lift1 . fmap Lift1 <$> m)
-#endif
 
-#ifdef LIFTED_FUNCTOR_CLASSES
 instance (Show1 f, Show1 m, Show a) => Show (FreeT f m a) where
-#else
-instance (Functor f, Show1 f, Functor m, Show1 m, Show a) => Show (FreeT f m a) where
-#endif
   showsPrec = showsPrec1
 
-#ifdef LIFTED_FUNCTOR_CLASSES
 instance (Read1 f, Read1 m) => Read1 (FreeT f m) where
   liftReadsPrec rp rl = go
     where
       goList = liftReadList rp rl
       go = readsData $ readsUnaryWith
         (liftReadsPrec (liftReadsPrec2 rp rl go goList) (liftReadList2 rp rl go goList))
         "FreeT" FreeT
-#else
-instance (Functor f, Read1 f, Functor m, Read1 m) => Read1 (FreeT f m) where
-  readsPrec1 d =  readParen (d > 10) $ \r ->
-    [ (FreeT (fmap lower1 . lower1 <$> m),t) | ("FreeT",s) <- lex r, (m,t) <- readsPrec1 11 s]
-#endif
 
-#ifdef LIFTED_FUNCTOR_CLASSES
 instance (Read1 f, Read1 m, Read a) => Read (FreeT f m a) where
-#else
-instance (Functor f, Read1 f, Functor m, Read1 m, Read a) => Read (FreeT f m a) where
-#endif
   readsPrec = readsPrec1
 
 instance (Functor f, Monad m) => Functor (FreeT f m) where
```

Comment:
We could go even further and simplify the `Monad m` constraint to `Functor m`, but doing so would require some code changes. I'll defer that to a future PR.